color.c

```diff
@@ -323,6 +323,10 @@ static int check_auto_color(int fd)
 {
 	static int color_stderr_is_tty = -1;
 	int *is_tty_p = fd == 1 ? &color_stdout_is_tty : &color_stderr_is_tty;
+	
+	if (git_env_bool("GIT_TEST_PRETEND_TTY", 0))
+		return 1;
+
```

Comment:
This would seem to me as a separate concern from the main part of this patch: I would recommend splitting it into two: a commit that introduces that `GIT_TEST_PRETEND_TTY` handling, and the next commit that uses `add-helper --show-help` and adds the test case that uses that environment variable.